nck/utils/date_handler.py

```diff
@@ -37,6 +37,21 @@ def __get_previous_month_dates(current_date: date) -> Tuple[date, date]:
 }
 
 
+def check_date_range_definition_conformity(start_date: date, end_date: date, date_range: str):
+    if start_date is None and end_date is None and date_range is None:
+        raise Exception(
+            "You must at least define a couple \
+                        start-date/end-date or a date-range"
+        )
+    elif (start_date is None and end_date is None) or (start_date is None or end_date is None):
```

Comment:
Hmm, not sure about this condition. Basically you want to make sure that you either have both start date and end date or none of them right ? Therefore you need to raise an exception for the cases where you have only one of the two which is None and the other one who is not None. Example of cases which will raise an exception : {start_date : None, end_date: 2020-06-01} or {start_date : 2020-06-01, end_date: None}. But here you will raise an exception for a case where both are None which can be a correct case if you only have a date range. Basically your condition needs to be an xor.

Reply:
xor on start-date/end-date but I may be wrong. Let me know what you think about it.

Reply:
Yes you were absolutely right, that was a mistake. I think I have taken into account what you said in the latest version of the branch.